### DISCUSSION TITLE
Long delay before displaying items  (Combobox)

### DISCUSSION THREAD

**Discussion by washburnit** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2724] [Q&A] [answered]
Hi,

I have a Combobox that is loaded with about 1300 items.
Items are read from a SQL table, loaded into a "class" with two elements and then assigned as "ItemSource" to the Combobox.
Apparently loading all the items is very fast but when I click on the combobox to open the list, it takes about 5/6 seconds to display the items.
Maybe I'm loading the combo in the wrong way, but the same "issue" occurs even if I load simple strings.
That is XAML and code behind:

`<ComboBox x:Name="cbCli" Style="{StaticResource MaterialDesignOutlinedComboBox}" 
                      mdix:HintAssist.Hint="Ricerca Cliente"  mdix:HintAssist.HintOpacity=".26"  IsEditable="True" 
                      Width="500" FontFamily="Century Gothic" FontSize="15" Margin="20,20,0,0" HorizontalAlignment="Left"/>
`
` 

           private class DatiCli
           {
                 public UInt32 IDc { get; set; }
                 public String ragSoc { get; set; }
           }

          //********* CODE TO LOAD THE COMBOBOX ***************//
           String str="";
           UInt32 IDc=0; 
           DatiCli DC= new DatiCli; 
           DatiCli lDC = new List<DatiCli>();

            while (PQuery.Read())
            {
                //ComboboxItem cbItem = new ComboboxItem();
                DC = new DatiCli();
                IDc = Convert.ToUInt32(PQuery["ID"]);
                str = PQuery["RAGIONESOCIALE"].ToString();
                DC.IDc = IDc;
                DC.ragSoc = str;
                lDC.Add(DC);                
            }

            cbCli.ItemsSource = lDC;
            cbCli.DisplayMemberPath = "ragSoc";
            cbCli.SelectedValuePath = "IDc";
`

I noticed that with a "standard" combobox (in a "non wpf" program) the combobox doesn't have any delay.
Thank you, regards.

Roberto

**Comment by Keboo** [comment, marked as answer]
Thanks Roberto, this sounds like a data virtualization issue. Looking over the code it appears that this library does not have virtualization enabled by default (this is a bug that should get fixed). 
In the meantime you should be able to [follow the virtualized ComboBoxes from the demo app](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/MainDemo.Wpf/ComboBoxes.xaml#L178-L182) to enabled virtualization.

```xaml
                <ComboBox
                    materialDesign:HintAssist.Hint="Virtualisation"
                    MinWidth="72"
                    ItemsSource="{Binding LongListToTestComboVirtualization}"
                    SelectedValue="{Binding SelectedValueOne}">

                    <ComboBox.SelectedItem>
                        <Binding
                            Path="SelectedValueOne"
                            Mode="TwoWay"
                            UpdateSourceTrigger="PropertyChanged">
                            <Binding.ValidationRules>
                                <demoAppDomain:NotEmptyValidationRule
                                    ValidatesOnTargetUpdated="True" />
                            </Binding.ValidationRules>
                        </Binding>
                    </ComboBox.SelectedItem>
                    
                    <ComboBox.ItemsPanel>
                        <ItemsPanelTemplate>
                            <VirtualizingStackPanel />
                        </ItemsPanelTemplate>
                    </ComboBox.ItemsPanel>
                </ComboBox>
```

**Reply by ElieTaillard** [reply]
Issue created: https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/issues/2725

**Reply by washburnit** [reply]
Thanks Keboo, your suggestion fixed the issue.
Regards,


Roberto
